src/app/utils.js

```diff
@@ -51,13 +51,14 @@ export const getModelStatusGroupData = (model) => {
       messages.push(app.status.info);
       return;
     }
-    Object.keys(app.units).forEach((unitId) => {
-      const unit = app.units[unitId];
+    const units = app.units || {}; // subordinates do not have units.
```

Comment:
You'll see this throughout. The old jujulib returned default values for returned datasets. This was incorrect and has been remedied in the new jujulib but this means we have to handle it in the application where necessary.